### ISSUE TITLE
Definitions

### ISSUE THREAD

**Issue by shazooer** [sunlightlabs/opendatapoliciesdecoded#4]
Create a glossary of multiple definitions of phrases
Take advantage of State Decoded is already tagging and identifying 
For terms with multiple definitions we'd like to know in what policies the definitions exist.

**Comment by jalbertbowden** [comment]
i've got one going for foia data, if you want to compare/contrast, etc.

**Comment by SLarrick** [comment]
Noting that it seems one can download a JSON of this on the downloads page.

**Comment by jalbertbowden** [comment]
most offer that functionality...i'm thinking one that is in the document, so you can actually use it while reading the content. something a little interactive

**Comment by SLarrick** [comment]
[dictionary.json.zip](https://github.com/sunlightlabs/opendatapoliciesdecoded/files/384290/dictionary.json.zip)

**Comment by SLarrick** [comment]
Seamus tells me we can change the character limit on the in-line definitions. Let's increase that limit a good amount. @loandy @shazooer

**Comment by SLarrick** [comment]
Here is a .csv of the above JSON. Seems like there are a lot of terms not captured?

https://docs.google.com/spreadsheets/d/19_zBQDPVxIpptsw2MpD2CblXk50R0t-Bx1p5iYuEnFo/edit#gid=2130684880

**Comment by waldoj** [comment]
Yup: http://docs.statedecoded.com/installation.html#definition-parsing

**Comment by SLarrick** [comment]
Thanks @waldoj! sorry for asking an obvious question!

@loandy @gregjd @shazooer it looks like the definitions pulled are character limited (so the definitions in our dictionary are incomplete both in that we are not capturing them all (at least it doesn't look like we are), and in that they are only partial/truncated definitions)... Let me know if this is easily fixable? I didn't see anything about a character limit in the documentation waldo just shared, but then again, I'm very likely missing lots of things (see my above obvious question).

**Comment by waldoj** [comment]
The State Decoded has no character limits for definitions.

**Comment by SLarrick** [comment]
Hmm. Guess the parser is only grabbing partial definitions then.

**Comment by SLarrick** [comment]
Relevant passage from http://docs.statedecoded.com/installation.html#definition-parsing: 

> If the terms within your legal code are not stored within double quotation marks, then extract_definitions() will need to be modified to be able to isolate those terms, such as locating them within <em></em> tags. For those rare legal codes that do not offset defined terms in any way (italics, quotation marks, or otherwise; e.g., Nebraska), modifications to the code will be necessary in order to isolate the term being defined.
> 
> The State Decode does its best to allow for the many terms and phrases employed by different legal codes, but this may need to be tuned to accommodate the legal code that you are parsing. Survey how it indicates the scope of definitions, and how it connects a term to its definition, and modify extract_definitions() to suit those circumstances.

@loandy @shazooer @gregjd Looks like we will need to modify ``extract_definitions()``  and should not expect for this to work seamlessly on roll out of V1.0 next week.

**Comment by waldoj** [comment]
For most legal codes, it's easy, because there are a few standard patterns for how definitions are structured. It'll be trickier here, because you're excerpting laws and policies from a bunch of places and combining them, so the patterns will be more widely divergent. But, really, you just need somebody to look through the policies, note which definition structures aren't being included, and append to `$linking_phrases`. You should probably leave `$scope_indicators` as an empty array, since you never want definitions to leak into other policies' definitions—that will ensure that every defined term has a scope that is limited to the current policy.
